tests/framework-tests/core/js/WebWorkerTests-Worker.js

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2015 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+/*jshint worker:true*/
+/*global fluid*/
+
+(function () {
+    "use strict";
+
+    importScripts(
+        "../../../../src/framework/core/js/jquery.standalone.js",
+        "../../../../src/framework/core/js/Fluid.js",
+        "../../../../src/framework/core/js/FluidDebugging.js",
+        "../../../../src/framework/core/js/FluidIoC.js",
+        "../../../../src/framework/core/js/DataBinding.js",
+        "../../../../src/framework/core/js/FluidPromises.js",
+        "../../../../src/framework/core/js/FluidRequests.js",
+        "../../../../src/framework/core/js/ModelTransformation.js",
+        "../../../../src/framework/core/js/ModelTransformationTransforms.js"
+    );
+
+    fluid.defaults("fluid.test.workerTestComponent", {
+        gradeNames: "fluid.modelComponent",
```

Comment:
Fault here - definition requires "autoInit" in order to function with the ancient framework. Also - both branches mysteriously use new() to construct the component, which does actually work, per the discussion we had last week, whereby "anomalous new" constructions will simply discard the initially constructed "this" object.